hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/handlers/TokenDissociateFromAccountHandler.java

```diff
@@ -144,21 +146,25 @@ public void handle(@NonNull final HandleContext context) {
 
                 if (tokenRelBalance > 0) {
                     validateFalse(token.tokenType() == TokenType.NON_FUNGIBLE_UNIQUE, ACCOUNT_STILL_OWNS_NFTS);
-                    // If the fungible token is NOT expired, then we throw an exception because we
-                    // can only dissociate tokens with a zero balance by this time in the code
-                    // @future('6864'): uncomment when token expiry is implemented
-                    // validateTrue(tokenIsExpired, TRANSACTION_REQUIRES_ZERO_TOKEN_BALANCES);
-
-                    // If the fungible common token is expired, we automatically transfer the
-                    // dissociating account's balance back to the token's treasury
-                    final var treasuryTokenRel = dissociation.treasuryTokenRel();
-                    if (treasuryTokenRel != null) {
-                        final var updatedTreasuryBalanceTokenRel = treasuryTokenRel.balance() + tokenRelBalance;
-                        treasuryBalancesToUpdate.add(treasuryTokenRel
-                                .copyBuilder()
-                                .balance(updatedTreasuryBalanceTokenRel)
-                                .build());
-                    }
+
+                    // Remove when token expiry is implemented
+                    throw new HandleException(TRANSACTION_REQUIRES_ZERO_TOKEN_BALANCES);
+                }
```

Comment:
`validateFalse` throws a `HandleException`. Why do we need to throw here ?

Reply:
you mean to add this `validateTrue(tokenIsExpired,TRANSACTION_REQUIRES_ZERO_TOKEN_BALANCES);`

Reply:
The logic for `tokenIsExpired` is not implemented yet, and it is always `false` for now. I didn't use a `validateTrue(balance  == 0, TRANSACTION_REQUIRES_ZERO_TOKEN_BALANCES)` here because this is already in a IF statement with that condition.

A few lines below there is a comment with `validateTrue(tokenIsExpired,TRANSACTION_REQUIRES_ZERO_TOKEN_BALANCES);`. It should be uncommented when `tokenIsExpired` is implemented and the `throw` should be deleted.